servers/src/test/java/tachyon/metrics/MetricsCountersTest.java

```diff
@@ -0,0 +1,367 @@
+import tachyon.thrift.FileInfo;
+import tachyon.thrift.NetAddress;
+import tachyon.underfs.UnderFileSystem;
+
+/**
+ * Unit tests for {@link tachyon.master.MasterSource}.
+ */
+public final class MetricsCountersTest {
+  private static final long TTLCHECKER_INTERVAL_MS = 0;
+  private static final TachyonURI NESTED_FILE_URI = new TachyonURI("/nested/test/file");
+  private static final TachyonURI ROOT_URI = new TachyonURI("/");
+  private static final TachyonURI ROOT_FILE_URI = new TachyonURI("/file");
+  private static final TachyonURI TEST_URI = new TachyonURI("/test");
+
+  private static final TachyonURI DIRECTORY_URI = new TachyonURI("/directory");
+  private static final TachyonURI MOUNT_URI =
+      new TachyonURI("/tmp/mount-" + System.currentTimeMillis());
+
+  private static CreateOptions sNestedFileOptions =
+      new CreateOptions.Builder(MasterContext.getConf()).setBlockSizeBytes(Constants.KB)
+          .setRecursive(true).build();
+
+  private BlockMaster mBlockMaster;
+  private FileSystemMaster mFileSystemMaster;
+  private long mWorkerId;
+
+  private UnderFileSystem mUfs = null;
+
+  Map<String, Counter> mCounters;
+
+  @Rule
+  public TemporaryFolder mTestFolder = new TemporaryFolder();
+
+  @Rule
+  public ExpectedException mThrown = ExpectedException.none();
+
+  @Before
+  public void before() throws Exception {
+    MasterContext.getConf().set(Constants.MASTER_TTLCHECKER_INTERVAL_MS,
+        String.valueOf(TTLCHECKER_INTERVAL_MS));
+    Journal blockJournal = new ReadWriteJournal(mTestFolder.newFolder().getAbsolutePath());
+    Journal fsJournal = new ReadWriteJournal(mTestFolder.newFolder().getAbsolutePath());
+    HeartbeatContext.setTimerClass(HeartbeatContext.MASTER_TTL_CHECK,
+        HeartbeatContext.SCHEDULED_TIMER_CLASS);
+
+    mBlockMaster = new BlockMaster(blockJournal);
+    mFileSystemMaster = new FileSystemMaster(mBlockMaster, fsJournal);
+
+    mBlockMaster.start(true);
+    mFileSystemMaster.start(true);
+
+    // set up worker
+    mWorkerId = mBlockMaster.getWorkerId(new NetAddress("localhost", 80, 81, 82));
+    mBlockMaster.workerRegister(mWorkerId, Arrays.asList("MEM", "SSD"),
+        ImmutableMap.of("MEM", (long) Constants.MB, "SSD", (long) Constants.MB),
+        ImmutableMap.of("MEM", (long) Constants.KB, "SSD", (long) Constants.KB),
+        Maps.<String, List<Long>>newHashMap());
+
+    MasterContext.reset();
+    mCounters = MasterContext.getMasterSource().getMetricRegistry().getCounters();
+
+    mUfs = UnderFileSystem.get(TachyonURI.SEPARATOR, MasterContext.getConf());
+  }
+
+  @Test
+  public void createFileTest() throws Exception {
+    mFileSystemMaster.create(ROOT_FILE_URI, sNestedFileOptions);
+
+    Assert.assertEquals(1, mCounters.get("CreateFileOps").getCount());
+    Assert.assertEquals(1, mCounters.get("FilesCreated").getCount());
+
+    // trying to create a file that already exist
+    try {
+      mFileSystemMaster.create(ROOT_FILE_URI, sNestedFileOptions);
```

Comment:
For all of the try/catch which expect an exception, you should add an Assert for when the exception is NOT thrown.